.lagoon.yml

```diff
@@ -16,3 +9,73 @@ environments:
         schedule: "0 * * * *"
         command: 'drush cron --root=/app'
         service: cli
+
+# Note regarding tasks in the scaffold.
+# Please avoid making scripts more than one or two lines, and avoid if/else logic.
+# When PaaS users or the support team modify this file, it should be easy to turn things on and off.
+
+tasks:
+  pre-rollout:
+    - run:
+        name: Ensure backup folder exists - not all current envs have one
+        command: mkdir -p /app/web/sites/default/files/private/backups
+        service: cli
+        shell: bash    
+    - run:
+        name: Snapshot the database and store
+        command: if [[ "$LAGOON_ENVIRONMENT_TYPE" = "production" ]]; then drush sql:dump --root=/app --gzip --result-file=/app/web/sites/default/files/private/backups/pre-deploy-dump.sql; fi
+        service: cli
+        shell: bash
+    - run:
+        name: Snapshot the config and store
+        command: if [[ $(drush cex --root=/app sync -y --quiet --destination "$TMP"/config; echo "$?") = 0 ]]; then tar -czf /app/web/sites/default/files/private/backups/pre-deploy-config.tar.gz "$TMP"/config --remove-files; fi
+        service: cli
+        shell: bash
+
+  post-rollout:
+    - run:
+        name: Correct legacy drush alias if necessary
+        command: sed -i "s/%%PROJECT_NAME%%/\${env.LAGOON_PROJECT}/g" /app/drush/sites/govcms.site.yml
+        service: cli
+        shell: bash
+    - run:
+        name: Ensure backup folder exists
+        command: mkdir -p /app/web/sites/default/files/private/backups
+        service: cli
+        shell: bash
+    - run:
+        name: If a new environment populate database from master
```

Comment:
We discussed that `govcms-deploy` does too much, because it was doing both platform-dependent (backups etc) and platform-independent (updb, cr, cim) steps. The solution there was to do exactly what you did above - putting platform-dependent steps into `.lagoon.yml`, but leaving platform-independent steeps in `govcms-deploy`.

The changes here as copying the logic in `govcms-deploy` (but not exactly). This makes CI deployment steps to be different to production. You want to have unified approach to define platform-independent steps, so that all changes would go into one place.